lib/pbench/server/api/resources/graphql_api.py

```diff
@@ -1,6 +1,87 @@
 import requests
 from flask_restful import Resource, abort
-from flask import request, make_response
+from flask import request, make_response, jsonify
+from pbench.server.api.resources.auth import auth
+from pbench.server.api.resources.graphql_schema import schema
+
+
+class UserMetadata(Resource):
+    """
+    Abstracted pbench API for handling user metadata by using graphql schema
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+
+    @auth.login_required()
```

Comment:
Hi Nikhil. I was just having a look a this PR to understand the server side better. so, what does this annotation mean? Does it check if user is authenticated or not? If yes, how does it do that?

Reply:
Hey @aquibbaig , So the decorator `@auth.login_required()` works by checking the auth token provided in the header with http request. We expect the auth header to be set like this `Authorization:   Bearer <JWT Token>` and a user get this token by making a post request to an API endpoint `/v1/session` (login process). Once the user gets the Auth token it has to be present is the header request in all the subsequent request where login is required. 

Now how this works internally is like the following
- We import our `auth` instance on line 4 from `auth.py` file 
- Inside our auth.py file we have an instance `auth` that is of type `HTTPTokenAuth` that takes `Bearer` argument
- In the same file there is a `verify_token` method that is defined with a decorator `@auth.verify_token` (note: this auth instance is same as the one used for login_required)
- So whenever a `login_required` is called it will invoke that auth instance's verify_token method which takes `auth_token` as an argument which was provided in the header as `Authorization Bearer <auth_token>`
- If `verify_token` returns true meaning if its a valid auth token the server will start to execute the post request